icmplib/sockets.py

```diff
@@ -170,9 +170,12 @@ def _bind_to_device(self, interface):
         # SO_BINDTODEVICE requires CAP_NET_RAW capability on Linux
         # Interface name must be encoded to bytes
         try:
+            # Try to get SO_BINDTODEVICE constant, fallback to 25 for compatibility
+            SO_BINDTODEVICE = getattr(socket, "SO_BINDTODEVICE", 25)
             self._sock.setsockopt(
                 socket.SOL_SOCKET,
                 25,  # SO_BINDTODEVICE
+                SO_BINDTODEVICE,
                 interface.encode(),
```

Comment:
Not part of this PR really, but we should raise an error if we use non-ascii characters in the interface name (encode() defaults to utf-8)
```suggestion
                interface.encode("ascii"),
```

Reply:
Updated